### PR TITLE
Feature/#63 cart test

### PR DESCRIPTION
**Pull request by pbimpisidis**

### CODE REVIEW

File: it/Cart.ps1 [hunk 1]
```diff
@@ -1,24 +1,119 @@
-function CreateCartItem($catItem) 
+function Create-CartItem
 {
+	Param
+	(
+		$Svc
+		,
+		$Name
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis $Name and $Svc will be mandatory as well, right?

File: it/Cart.ps1 [hunk 2]
```diff
@@ -1,24 +1,119 @@
...
+	
+	#save changes
+	$svc.Core.UpdateObject($cartItem)
+	$result = $svc.core.SaveChanges();	
+				
+	#get the updated Cart Item 
+	$updatedCartItem = $svc.Core.CartItems.AddQueryOption('$filter', $query) | select;
+	
+	#ASSERT - update
+	if ($Name){$bin = $updatedCartItem.Name | Should Be $Name;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis please format for better readability

```
if ($Name)
{
	$bin = $updatedCartItem.Name | Should Be $Name;
}
```

File: it/Cart.ps1 [hunk 3]
```diff
@@ -1,24 +1,119 @@
...
+	if ($Description) {$bin = $updatedCartItem.Description | Should Be $Description;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis same as above

File: it/Cart.ps1 [hunk 4]
```diff
@@ -1,24 +1,119 @@
...
+	if ($Quantity) {$bin = $updatedCartItem.Quantity | Should Not Be $Quantity;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis same as above

File: it/Cart.ps1 [hunk 5]
```diff
@@ -1,24 +1,119 @@
...
+		,
+		$Quantity
+	)
+	
+	#get the CartItem 
+	$query = "Id eq {0}" -f $Id;
+	$cartItem = $svc.Core.CartItems.AddQueryOption('$filter', $query) | select;
+	
+	#Set new name and description for the Cart Item
+	if ($Name){$cartItem.Name = $Name;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis format if statement according allman coding style

File: it/Cart.ps1 [hunk 6]
```diff
@@ -1,24 +1,119 @@
...
+	if ($Description) {$cartItem.Description = $Description;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis format if statement according allman coding style

File: it/Cart.ps1 [hunk 7]
```diff
@@ -1,24 +1,119 @@
...
+	if ($Quantity) {$cartItem.Quantity = $Quantity;}
```
Comment (rufer7) [inline it/Cart.ps1]: @pbimpisidis format if statement according allman coding style

**Comment by pbimpisidis** [comment]
all fixed

File: it/Cart.Tests.ps1 [hunk 8]
```diff
@@ -11,181 +13,152 @@ function Stop-Pester($message = "EMERGENCY: Script cannot continue.")
...
+			$cartItemId = $cartItem1.Id;
+			$cartId = $cartItem1.CartId;
+			
+			#ACT create second cart item, same as the previous
+			try
 			{
-				$svc.Core.SaveChanges();
-			} catch 
+				$cartItem2 = Create-CartItem -svc $svc -Name $cartItemName -CatalogueItemId $catalogueItemId;
+			}
```
Comment (dfch) [inline it/Cart.Tests.ps1]: @pbimpisidis use Pester.Assertions and `ThrowDataServiceClientException` with an expected `StatusCode` (see test examples in repo)

**Comment by rufer7** [comment]
@pbimpisidis ok now?

**Pull request merged**